python.d/python_modules/base.py

```diff
@@ -468,11 +512,15 @@ def _get_raw_data(self):
                 self.sock = None
                 return None
 
-        data = sock.recv(1024)
+        data = sock.recv(2)
         try:
             while True:
-                buf = sock.recv(1024)
-                if not buf:
+                try:
+                    buf = sock.recv(1024, 0x40)  # get 1024 bytes in NON-BLOCKING mode
```

Comment:
Sorry, I didn't get this. In order to assume you received a response, you need to timeout?